lib/db.c

```diff
@@ -74,6 +74,14 @@ static const uint32_t GRN_TABLE_PAT_KEY_CACHE_SIZE = 1 << 15;
       grn_string_get_normalized(ctx, nstr, &key, &key_size, NULL);\
       block\
       grn_obj_close(ctx, nstr);\
+    } else {                   \
+      if (key && key_size > 0) {\
+        ERR(GRN_INVALID_ARGUMENT,\
+            "failed to open string by key:<%.*s> nstr:<(nil)>", key_size, (char *)key); \
```

Comment:
Could you remove a needless space before `\`?

Reply:
`[key][normalize] failed to normalize: <%.*s>", (int)key_size, (const char *)key)` is better.
Including table information will better. We can use `DEFINE_NAME()` in `lib/ii.c` for this.
(Generalizing `DEFINE_NAME()` should be done in separated pull request.)